### PR TITLE
chore: add theorem for `bitvec_152`

### LINKED ISSUE #602: b * BitVec.allOnes w = -b
`bitvec_152` in AliveStatements requires this proof to be automatically solved.

### PR DESCRIPTION
**Pull request by luisacicolini**
PR to solve [issue#602](https://github.com/opencompl/lean-mlir/issues/602) (`b * BitVec.allOnes w = -b`)

### CODE REVIEW

File: SSA/Projects/InstCombine/ForLean.lean [hunk 1]
```diff
@@ -528,6 +528,11 @@ theorem one_shiftLeft_mul {x y : BitVec w} :
     1#w <<< x.toNat * y = y <<< x.toNat := by
   simp [←BitVec.mul_twoPow_eq_shiftLeft, BitVec.mul_comm]
 
+@[simp]
+theorem neg_one_mul {x : BitVec w} :
+    x.mul (-1) = (0#w).sub x := by
+  simp [BitVec.mul_comm]
+
 end BitVec
```
Comment (tobiasgrosser) [inline SSA/Projects/InstCombine/ForLean.lean]:
```suggestion 

@[simp]
theorem mul_allOnes {x : BitVec w} :
    x * BitVec.allOnes w = -x := by
  simp [← BitVec.negOne_eq_allOnes]

end BitVec
```

Comment (tobiasgrosser) [reply SSA/Projects/InstCombine/ForLean.lean]: To be able to put a `@[simp]` decorator, we need to make sure that the RHS is strictly simpler than the RHS. I feel this is not true for the theorem you choose, but there is another theorem after `alive_auto` has run that where a simp rewrite will lead to a clear simplification. I added this for you.

**Review by tobiasgrosser — APPROVED** [review]

File: SSA/Projects/InstCombine/ForLean.lean (hunk 1 above)
Comment (tobiasgrosser) [reply SSA/Projects/InstCombine/ForLean.lean]: If my comments make sense, you can commit to the suggestion and merge the PR.

Thank you for working on this. This really helps to develop a joint language.

Comment (luisacicolini) [reply SSA/Projects/InstCombine/ForLean.lean]: thanks! this also clarifies the use of `@[simp]`

**Pull request merged**